### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by EnoF**
Major UPGRADE!!!!

please review with care.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32300645%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32300660%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32301756%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32301764%22%2C%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32301958%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20a1cc82b21ebb3b0f1836ca6a45d61b364015c793%20app/core/dao/dao.ts%2050%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32300660%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5E%20also%20here%22%2C%20%22created_at%22%3A%20%222015-06-12T09%3A10%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/core/dao/dao.ts%3AL1-74%22%7D%2C%20%22Pull%20a1cc82b21ebb3b0f1836ca6a45d61b364015c793%20app/widgets/call/src/callForm.ts%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32301756%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22We%20can%20remove%20the%20comments.%22%2C%20%22created_at%22%3A%20%222015-06-12T09%3A27%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/widgets/call/src/callForm.ts%3AL1-17%22%7D%2C%20%22Pull%20a1cc82b21ebb3b0f1836ca6a45d61b364015c793%20app/core/dao/dao.ts%2042%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Jumio/con-rest/pull/80%23discussion_r32300645%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22don%27t%20we%20have%20the%20case%20where%20we%20have%20to%20send%20multiple%20files%3F%22%2C%20%22created_at%22%3A%20%222015-06-12T09%3A09%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22On%20the%20client%20we%20do%20not%28for%20now%29%20serverside%20is%20untouched.%20So%20that%20means%20we%20support%20that%20still%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-06-12T09%3A28%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22uuuups%20%3AD%22%2C%20%22created_at%22%3A%20%222015-06-12T09%3A31%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/core/dao/dao.ts%3AL1-74%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull a1cc82b21ebb3b0f1836ca6a45d61b364015c793 app/core/dao/dao.ts 42'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/80#discussion_r32300645'>File: app/core/dao/dao.ts:L1-74</a></b>
- <a href='https://github.com/dominikkukacka'></a> don't we have the case where we have to send multiple files?
- <a href='https://github.com/EnoF'></a> On the client we do not(for now) serverside is untouched. So that means we support that still :)
- <a href='https://github.com/dominikkukacka'></a> uuuups :D
- [ ] <a href='#crh-comment-Pull a1cc82b21ebb3b0f1836ca6a45d61b364015c793 app/core/dao/dao.ts 50'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/80#discussion_r32300660'>File: app/core/dao/dao.ts:L1-74</a></b>
- <a href='https://github.com/dominikkukacka'></a> ^ also here
- [ ] <a href='#crh-comment-Pull a1cc82b21ebb3b0f1836ca6a45d61b364015c793 app/widgets/call/src/callForm.ts 1'></a> &nbsp;<b><a href='https://github.com/Jumio/con-rest/pull/80#discussion_r32301756'>File: app/widgets/call/src/callForm.ts:L1-17</a></b>
- <a href='https://github.com/dominikkukacka'></a> We can remove the comments.

<a href='https://www.codereviewhub.com/Jumio/con-rest/pull/80?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Jumio/con-rest/pull/80?approve=1'></a>&nbsp;<a href='https://github.com/Jumio/con-rest/pull/80'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by EnoF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Jumio/con-rest/pull/80'>Refresh</a>.<a href='#crh-update-none'></a>

File: app/core/dao/dao.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        config.headers['Content-Type'] = undefined;
+      }
+      return this.$http(config);
+    }
+
+    get(url: string, params: Object) {
+      return this.execute(this.GET, url, null, params);
+    }
+
+    post(url: string, data: Object, fileBoundaryName?: string, file?: any) {
```
Comment (dominikkukacka) [inline app/core/dao/dao.ts]: don't we have the case where we have to send multiple files?

File: app/core/dao/dao.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      if (!!fileBoundaryName && !!file) {
+        return this.execute(this.POST, url, this.createFormData(data, fileBoundaryName, file));
+      } else {
+        return this.execute(this.POST, url, data);
+      }
+    }
+
+    put(url: string, data: Object, fileBoundaryName?: string, file?: any) {
```
Comment (dominikkukacka) [inline app/core/dao/dao.ts]: ^ also here

File: app/widgets/call/src/callForm.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (dominikkukacka) [inline app/widgets/call/src/callForm.ts]: We can remove the comments.

File: app/core/dao/dao.ts (hunk 1 above)
Comment (EnoF) [inline app/core/dao/dao.ts]: On the client we do not(for now) serverside is untouched. So that means we support that still :)

Comment (dominikkukacka) [inline app/core/dao/dao.ts]: uuuups :D

**Pull request merged**